RGPosadas/LitterRally
# TASK-19: Multiple Registration Options Back-end

This PR implements the backend of Multiple Registration Options #73.

**Steps to Test:**
1. Go to the [slack](https://bug-bytes.slack.com/archives/CS5P4NL5Q/p1606083803001700) and update your ```.env``` file.
1. Install dependencies using ```npm i```.
1. Start the server using ```nx serve mull-api```.
1. In a browser, go to ```localhost:3333/api/auth/google``` , ```localhost:3333/api/auth/facebook```, ```localhost:3333/api/auth/twitter```
1. Follow the instructions for each service.
1. You should get a returned user after a successful OAuth validation.

**Notes:**
The accounts on the slack link can be used to test if you're not comfortable using your personal accounts.
There is a chance that Twitter OAuth might not work. If that is the case, reach out to me on slack.

Facebook requires me to add anyone who wants to test as a developer on the app. I will try inviting everyone but if it doesn't work, try logging in to the Facebook Developer account and adding yourself as a developer. (This is caused by the app being in development mode)
